packages/zk-transfer-web/src/App.tsx

```diff
@@ -1,57 +1,52 @@
 import React, { useState, useEffect } from "react";
 import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
-
+import { Map } from "immutable";
 import Homepage from './views/Home';
 import MakeTransaction from './views/MakeTransaction';
 import Democracy from './components/context/Democracy';
 import { getScreenName, setScreenName, generateScreenName, getZkTokenName } from "./util";
 import { makeApi } from "./libs/api";
-import { Note } from "./libs/types";
+import { Note, TokenAddress } from "./libs/types";
 
 type AppProp = {
   demo: any
 };
 
 function App({ demo } : AppProp) {
-  const [state, setState] = useState({
-    chainId: demo.chainId,
+  const [state, setState]: [any, any] = useState({
+    chainId: null,
     screenName: "",
-    ZKToken: new Map(),
+    ZKTokens: Map(),
   });
 
   useEffect(() => {
-    const { chainId } = state;
-
-    if (!chainId) {
+    if (!state.chainId)
       demo.clientInit()
         .then(async () => {
           console.groupCollapsed('Client Initialized');
-          const api = await makeApi(demo);
-          api.zkTokens.mapEntries(([tokenName, token]) => {
-            const shortTokenName = getZkTokenName(tokenName);
-            const tokenAddress: string = token.get("deployAddress");
+          const { zkTokens, thisAddressNotes } = await makeApi(demo);
+          const tokenAndNotes = zkTokens.mapEntries(([ tokenName, token]) => {
```

Comment:
a better name for `tokenAndNotes` is `tradeSymbolToNotes` or `notesMap`,
with a comment that tells us what the key is and the value is. like
```
// map of tradeSymbol (a string) to a NoteList

Reply:
I didn't add the comment because Typescript can tell the type for the `key` and `value`